.github/workflows/rust.yml

```diff
@@ -22,32 +22,37 @@ jobs:
         with:
           python-version: "3.11.0"
       - uses: Swatinem/rust-cache@v2
-      - name: Build
-        run: |
-          rustup update stable
-          cargo build --verbose
-      - name: Run tests
-        run: cargo test --features large_thread --verbose --all
-      - uses: actions-rs/cargo@v1
-        with:
-          command: clippy
-          args: -- -D warnings
-  test-win:
-    runs-on: windows-latest
+      - run: rustup update stable
+      - name: Compile
+        run: cargo test --features large_thread --all --verbose --no-run
+      - name: Run tests (Windows)
+        if: runner.os == 'Windows'
+        # HACK: The cause is unknown, but on windows, the exit code is 1 even if tests are successful.
+        run: cargo test --features large_thread --features pre-commit --all --verbose
+      - name: Run tests (Other OS)
+        if: matrix.os != 'Windows'
+        run: cargo test --features large_thread --all --verbose
+
+  build-check:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-python@v4
-        with:
-          python-version: "3.11.0"
       - uses: Swatinem/rust-cache@v2
-      - name: Build
-        run: |
-          rustup update stable
-          cargo build --verbose
-      - name: Run tests
-        # HACK: The cause is unknown, but on windows, the exit code is 1 even if tests are successful.
-        run: cargo test --features large_thread --features pre-commit --verbose --all
-      - uses: actions-rs/cargo@v1
+      - run: rustup update stable
+      - run: cargo build --all --all-targets --verbose --release
```

Comment:
This may be excessive.

Reply:
Hmm, it might be better to have a separate issue on this.

Reply:
Is the `--release` flag necessary? I think it would be faster without it.

Reply:
I think this level of CI time is acceptable, so I will assume it is not so important now.